### ISSUE TITLE
[Feature]: Evaluate multiple ngram speculations in speculative decoding

### ISSUE THREAD

**Issue by chenglu66** [vllm-project/vllm#6785]
### 🚀 The feature, motivation and pitch

 During the ngram-spec-decode stage, I've always had a question: In RAG, there isn't just one document relevant to the answer; why don't we first let the large model generate 3 tokens, and then take all possible results in the N-gram?

In simpler terms, imagine you're looking for an object in several rooms but can only carry three things at once. You might want to pick up some important items now so you won't forget them when carrying more stuff later. This way, you make sure your search is efficient and effective.

**Comment by cadedaniel** [comment]
You can do this, and in fact this is something we want to do in vLLM. The primary tradeoff is that verification of additional speculative tokens costs FLOPs, so if you don't pick carefully you can end up hurting overall performance (especially as number of concurrent requests grows). See https://github.com/vllm-project/vllm/issues/4565 for more details here.

In terms of implementing this in vLLM, we need https://github.com/vllm-project/vllm/issues/3960 first.
